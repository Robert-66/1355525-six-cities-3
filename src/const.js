export const MapSettings = {
  ZOOM: 12,
  ICON_URL: `img/pin.svg`,
  ICON_SIZE: [27, 39]
};
