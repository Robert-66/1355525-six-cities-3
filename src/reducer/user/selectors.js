export function getAuthorizationStatus(state) {
  return state.user.authorizationStatus;
}
