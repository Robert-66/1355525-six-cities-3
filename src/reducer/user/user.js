import {extend} from '../../utils';

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  email: ``,
};

const ActionType = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const ActionCreator = {
  auth: (authInfo) => {
    return {
      type: ActionType.AUTH,
      payload: authInfo,
    };
  },
  noAuth: () => {
    return {
      type: ActionType.NO_AUTH,
    };
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.AUTH:
      return extend(state, {authorizationStatus: AuthorizationStatus.AUTH, email: action.payload.email});
    case ActionType.NO_AUTH:
      return initialState;
  }

  return state;
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then((res) => {
        dispatch(ActionCreator.auth(res.data));
      });
  },
  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.email,
      password: authData.password,
    })
      .then((res) => {
        dispatch(ActionCreator.auth(res.data));
      });
  },
};


export {
  ActionCreator,
  ActionType,
  AuthorizationStatus,
  Operation,
  reducer,
};
