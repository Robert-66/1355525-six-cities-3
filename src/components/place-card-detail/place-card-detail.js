import React from 'react';
import {connect} from 'react-redux';
import {getCurrentCityLocation, getOffers} from '../../reducer/selectors';
import {extend} from '../../utils';
import PropTypes from 'prop-types';
import Reviews from '../reviews/reviews';
import Map from '../map/map';
import mockReviews from '../../mocks/reviews';
import {reviewType} from '../../types/reviews-types';
import {offerType} from '../../types/offers-types';
import PlaceCardList from '../place-card-list/place-card-list';

function PlaceCardDetail(props) {
  const {
    offer,
    offersNearby,
    offersNearbyMap,
    currentCityLocation,
    reviews,
  } = props;
  const {
    images,
    name,
    description,
    isPremium,
    type,
    bedrooms,
    maxAdults,
    price,
    rating,
    goods,
    host,
    id,
  } = offer.data;
  return (
    <main className="page__main page__main--property">
      {Object.keys(offer.data).length > 0 && (
        <>
          <section className="property">
            <div className="property__gallery-container container">
              <div className="property__gallery">
                {images.map((image, index) => (
                  <div key={`${image}-${index}`} className="property__image-wrapper">
                    <img className="property__image" src={image} alt={`Photo ${type}`} />
                  </div>
                ))}
              </div>
            </div>
            <div className="property__container container">
              <div className="property__wrapper">
                {isPremium && (
                  <div className="property__mark">
                    <span>Premium</span>
                  </div>
                )}
                <div className="property__name-wrapper">
                  <h1 className="property__name">
                    {name}
                  </h1>
                  <button className="property__bookmark-button button" type="button">
                    <svg className="property__bookmark-icon" width="31" height="33">
                      <use xlinkHref="#icon-bookmark" />
                    </svg>
                    <span className="visually-hidden">To bookmarks</span>
                  </button>
                </div>
                <div className="property__rating rating">
                  <div className="property__stars rating__stars">
                    <span style={{width: `${rating * 100 / 5}%`}} />
                    <span className="visually-hidden">Rating</span>
                  </div>
                  <span className="property__rating-value rating__value">{rating}</span>
                </div>
                <ul className="property__features">
                  <li className="property__feature property__feature--entire">
                    {type === `room` ? `Private Room` : type[0].toUpperCase() + type.slice(1)}
                  </li>
                  <li className="property__feature property__feature--bedrooms">
                    {bedrooms} Bedrooms
                  </li>
                  <li className="property__feature property__feature--adults">
                    Max {maxAdults} adults
                  </li>
                </ul>
                <div className="property__price">
                  <b className="property__price-value">&euro;{price}</b>
                  <span className="property__price-text">&nbsp;night</span>
                </div>
                <div className="property__inside">
                  <h2 className="property__inside-title">What&apos;s inside</h2>
                  <ul className="property__inside-list">
                    {goods.map((good, index) => (
                      <li key={`${good}-${index}`} className="property__inside-item">
                        {good}
                      </li>
                    ))}
                  </ul>
                </div>
                <div className="property__host">
                  <h2 className="property__host-title">Meet the host</h2>
                  <div className="property__host-user user">
                    <div className={`property__avatar-wrapper ${host.isPro ? `property__avatar-wrapper--pro` : ``} user__avatar-wrapper`}>
                      <img className="property__avatar user__avatar" src={host.avatarUrl} width="74" height="74" alt="Host avatar" />
                    </div>
                    <span className="property__user-name">
                      {host.name}
                    </span>
                    {host.isPro && (
                      <span className="property__user-status">Pro</span>
                    )}
                  </div>
                  <div className="property__description">
                    <p className="property__text">
                      {description}
                    </p>
                  </div>
                </div>
                <Reviews reviews={reviews} className="property__reviews" />
              </div>
            </div>
            <Map className="property__map" offers={offersNearbyMap} city={currentCityLocation} hoverOfferId={id} />
          </section>
          <div className="container">
            <section className="near-places places">
              <h2 className="near-places__title">Other places in the neighbourhood</h2>
              <PlaceCardList
                offers={offersNearby}
                className="near-places__list"
                classNamePlaceCard="near-places__card"
                classNamePlaceCardImageWrapper="near-places__image-wrapper"
              />
            </section>
          </div>
        </>
      )}
      {offer.isError && (
        <div className="container">
          <h1>Что-то пошло не так :(</h1>
        </div>
      )}
    </main>
  );
}

PlaceCardDetail.propTypes = {
  offer: PropTypes.shape({
    data: PropTypes.shape({
      city: PropTypes.shape({
        name: PropTypes.string.isRequired,
        location: PropTypes.shape({
          latitude: PropTypes.number.isRequired,
          longitude: PropTypes.number.isRequired,
          zoom: PropTypes.number.isRequired,
        }).isRequired,
      }),
      images: PropTypes.arrayOf(PropTypes.string),
      name: PropTypes.string,
      description: PropTypes.string,
      isPremium: PropTypes.bool,
      type: PropTypes.string,
      bedrooms: PropTypes.number,
      maxAdults: PropTypes.number,
      price: PropTypes.number,
      rating: PropTypes.number,
      goods: PropTypes.arrayOf(PropTypes.string),
      host: PropTypes.shape({
        name: PropTypes.string.isRequired,
        isPro: PropTypes.bool.isRequired,
        avatarUrl: PropTypes.string.isRequired,
      }),
      id: PropTypes.number.isRequired,
    }).isRequired,
    isLoading: PropTypes.bool.isRequired,
    isError: PropTypes.bool.isRequired,
  }).isRequired,
  offersNearby: PropTypes.arrayOf(offerType).isRequired,
  offersNearbyMap: PropTypes.arrayOf(offerType).isRequired,
  currentCityLocation: PropTypes.arrayOf(PropTypes.number),
  reviews: PropTypes.arrayOf(reviewType).isRequired,
};

function mapStateToProps(state, props) {
  let offer = extend({}, getOffers(state));
  const data = offer.data.find((item) => item.id === Number(props.match.params.id));
  offer.data = data ? data : {};
  let offersNearby = [];
  let offersNearbyMap = [];

  if (Object.keys(offer.data).length) {
    offersNearby = getOffers(state).data
      .filter((item) => item.city.name === offer.data.city.name && item.id !== offer.data.id)
      .slice(0, 3);
    offersNearbyMap = [...offersNearby];
    offersNearbyMap.push(offer.data);
  }

  const reviews = mockReviews
      .slice(0, 10)
      .sort((a, b) => Date.parse(b.date) - Date.parse(a.date));

  return {
    offer,
    offersNearby,
    offersNearbyMap,
    currentCityLocation: getCurrentCityLocation(state),
    reviews
  };
}

export {PlaceCardDetail};
export default connect(mapStateToProps)(PlaceCardDetail);
