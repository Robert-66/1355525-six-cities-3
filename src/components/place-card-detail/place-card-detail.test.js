import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import {CityNames} from '../../const';
import {AuthorizationStatus} from '../../reducer/user/user';
import {PlaceCardDetail} from './place-card-detail';

const mockStore = configureStore([]);

const mocks = {
  offer: {
    data: {
      city: {
        name: CityNames.AMSTERDAM,
        location: {
          latitude: 52.37454,
          longitude: 4.897976,
          zoom: 13
        }
      },
      description: `description`,
      bedrooms: 2,
      maxAdults: 3,
      goods: [`Air conditioning`, `Washer`, `Dishwasher`],
      images: [`img/image1.jpg`, `img/image2.jpg`],
      isFavorite: false,
      previewImage: `img/apartment-01.jpg`,
      isPremium: false,
      type: `apartment`,
      id: 11,
      name: `Apartments Prinsengracht`,
      price: 120,
      coords: [52.3909553943508, 4.85309666406198],
      rating: 4.3,
      host: {
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`,
      },
    },
    isLoading: false,
    isError: false,
  },
  offersNearby: [
    {
      city: {
        name: CityNames.AMSTERDAM,
        location: {
          latitude: 52.37454,
          longitude: 4.897976,
          zoom: 13
        }
      },
      previewImage: `img/apartment-01.jpg`,
      isPremium: true,
      type: `apartment`,
      id: 21,
      name: `Lovely Studio With Canal Views`,
      price: 99,
      coords: [52.369553943508, 4.85309666406198],
      rating: 3.1,
    },
    {
      city: {
        name: CityNames.AMSTERDAM,
        location: {
          latitude: 52.37454,
          longitude: 4.897976,
          zoom: 13
        }
      },
      previewImage: `img/apartment-01.jpg`,
      isPremium: true,
      type: `apartment`,
      id: 31,
      name: `Zandberg - Canal view apartments`,
      price: 140,
      coords: [52.3909553943508, 4.929309666406198],
      rating: 5,
    },
    {
      city: {
        name: CityNames.AMSTERDAM,
        location: {
          latitude: 52.37454,
          longitude: 4.897976,
          zoom: 13
        }
      },
      previewImage: `img/apartment-01.jpg`,
      isPremium: false,
      type: `room`,
      id: 41,
      name: `Bright & new apartment with canal view`,
      price: 125,
      coords: [52.3809553943508, 4.939309666406198],
      rating: 4.9,
    },
  ],
  reviews: [
    {
      comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      date: `2019-05-08T14:13:56.569Z`,
      id: 1,
      rating: 3,
      user: {
        avatarUrl: `img/1.png`,
        name: `Max`,
      }
    },
    {
      comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      date: `2017-08-08T14:13:56.569Z`,
      id: 2,
      rating: 5,
      user: {
        avatarUrl: `img/2.png`,
        name: `Anna`,
      }
    },
  ],
  reviewsForm: {
    data: [],
    isLoading: false,
    isError: false,
  },
  currentCityLocation: [52.37454, 4.897976],
};
mocks.offersNearbyMap = [mocks.offer.data, ...mocks.offersNearby];

it(`Should PlaceCardDetail render correctly`, () => {
  const store = mockStore({
    user: {
      authorizationStatus: AuthorizationStatus.AUTH,
    },
    data: {
      reviewsForm: mocks.reviewsForm,
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <PlaceCardDetail
              offer={mocks.offer}
              offersNearby={mocks.offersNearby}
              offersNearbyMap={mocks.offersNearbyMap}
              currentCityLocation={mocks.currentCityLocation}
              reviews={mocks.reviews}
              onChangeFavoriteStatus={() => {}}
            />
          </BrowserRouter>
        </Provider>,
        {
          createNodeMock: () => {
            return document.createElement(`section`);
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
