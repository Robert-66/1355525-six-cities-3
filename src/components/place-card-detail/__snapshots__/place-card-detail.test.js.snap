// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PlaceCardDetail render correctly 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo apartment"
            className="property__image"
            src="img/image1.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo apartment"
            className="property__image"
            src="img/image2.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            Apartments Prinsengracht
          </h1>
          <button
            className="property__bookmark-button button"
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "86%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            4.3
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            Apartment
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            2
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            3
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            120
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Air conditioning
            </li>
            <li
              className="property__inside-item"
            >
              Washer
            </li>
            <li
              className="property__inside-item"
            >
              Dishwasher
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="img/avatar-angelina.jpg"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              Angelina
            </span>
            <span
              className="property__user-status"
            >
              Pro
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              description
            </p>
          </div>
        </div>
        <section
          className="reviews property__reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews · 
            <span
              className="reviews__amount"
            >
              2
            </span>
          </h2>
          <ul
            className="reviews__list"
          >
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    src="img/1.png"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "60%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                </p>
                <time
                  className="reviews__time"
                  dateTime="2019-05-08"
                >
                  May 2019
                </time>
              </div>
            </li>
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    src="img/2.png"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                </p>
                <time
                  className="reviews__time"
                  dateTime="2017-08-08"
                >
                  August 2017
                </time>
              </div>
            </li>
          </ul>
          <form
            action="#"
            className="reviews__form form"
            onSubmit={[Function]}
          >
            <label
              className="reviews__label form__label"
              htmlFor="review"
            >
              Your review
            </label>
            <div
              className="reviews__rating-form form__rating"
            >
              <input
                checked={false}
                className="form__rating-input visually-hidden"
                disabled={false}
                id="5-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={5}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="5-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked={false}
                className="form__rating-input visually-hidden"
                disabled={false}
                id="4-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={4}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="4-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked={false}
                className="form__rating-input visually-hidden"
                disabled={false}
                id="3-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={3}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="3-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked={false}
                className="form__rating-input visually-hidden"
                disabled={false}
                id="2-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={2}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="2-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked={false}
                className="form__rating-input visually-hidden"
                disabled={false}
                id="1-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={1}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="1-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
            </div>
            <textarea
              className="reviews__textarea form__textarea"
              disabled={false}
              id="review"
              name="review"
              onChange={[Function]}
              placeholder="Tell how was your stay, what you like and what can be improved"
              value=""
            />
            <div
              className="reviews__button-wrapper"
            >
              <p
                className="reviews__help"
              >
                To submit review please make sure to set 
                <span
                  className="reviews__star"
                >
                  rating
                </span>
                 and describe your stay with at least 
                <b
                  className="reviews__text-amount"
                >
                  50 characters
                </b>
                .
              </p>
              <button
                className="reviews__submit form__submit button"
                disabled={true}
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
    <section
      className="map property__map"
    />
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="places__list near-places__list"
      >
        <article
          className="place-card near-places__card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="place-card__image-wrapper near-places__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="img/apartment-01.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  99
                </b>
                 
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button"
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "62%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                className="place-card__name-link"
                href="/offer/21"
                onClick={[Function]}
              >
                Lovely Studio With Canal Views
              </a>
            </h2>
            <p
              className="place-card__type"
            >
               
              Apartment
            </p>
          </div>
        </article>
        <article
          className="place-card near-places__card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="place-card__image-wrapper near-places__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="img/apartment-01.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  140
                </b>
                 
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button"
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                className="place-card__name-link"
                href="/offer/31"
                onClick={[Function]}
              >
                Zandberg - Canal view apartments
              </a>
            </h2>
            <p
              className="place-card__type"
            >
               
              Apartment
            </p>
          </div>
        </article>
        <article
          className="place-card near-places__card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="place-card__image-wrapper near-places__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="img/apartment-01.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  125
                </b>
                 
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button"
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "98.00000000000001%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                className="place-card__name-link"
                href="/offer/41"
                onClick={[Function]}
              >
                Bright & new apartment with canal view
              </a>
            </h2>
            <p
              className="place-card__type"
            >
               
              Private Room
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>
</main>
`;
